Mathlib/RingTheory/Flat/Stability.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Christian Merten
+-/
+import Mathlib.RingTheory.Flat.Basic
+import Mathlib.LinearAlgebra.TensorProduct.Tower
+
+/-!
+# Flatness is stable under composition and base change
+
+We show that flatness is stable under composition and base change. The latter is not formalized yet.
+
+## Main theorems
+
+* `Module.Flat.comp`: if `S` is a flat `R`-algebra and `M` is a flat `S`-module,
+                      then `M` is a flat `R`-module
+
+## TODO
+
+* Show that flatness is stable under base change, i.e. if `S` is any `R`-algebra and `M` is a flat
+  `R`-module, then `M ⊗[R] S` is a flat `S`-module.
+
+-/
+
+universe u v w t
+
+open Function (Injective Surjective)
+
+open LinearMap (lsmul rTensor lTensor)
+
+open TensorProduct
+
+namespace Module.Flat
+
+variable (R : Type u) (S : Type v) (M : Type w) (N : Type t)
+  [CommRing R] [CommRing S] [Algebra R S]
+  [AddCommGroup M] [Module R M] [Module S M] [IsScalarTower R S M]
+
```

Comment:
Could you please write a little module docstring that explains the proof strategy?

Reply:
Done in the latest commit.